autopaho/auto.go

```diff
@@ -207,8 +207,13 @@ func NewConnection(ctx context.Context, cfg ClientConfig) (*ConnectionManager, e
 			if cli == nil {
 				break mainLoop // Only occurs when context is cancelled
 			}
+			r := c.cli.Router // Get a reference to the previous client's Router
```

Comment:
I guess something like this is needed if `ClientConfig.Router` is `nil` - good catch; I had not considered that the temporary router would be thrown away in that case. However I wonder if a better solution would be to change `NewConnection` to do:

```
if cfg.Router == nil { // Ensure router will survive reconnections
   cfg.Router = paho.NewStandardRouter()
}
```

I'm not going to hold up your PR because of this but it feels cleaner to me and avoids creating then throwing away a router (I am being picky but think there is a potential, very unlikely to actually occur, race with the first packet from the broker and have been bitten by similar things in the past).